server/channels/api4/post.go

```diff
@@ -1107,6 +1110,83 @@ func unacknowledgePost(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func moveThread(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
+1 on this, seems we don't want to move from a read only/ archived channel?

Reply:
There are definitely some intricacies when you stray outside the "golden path". Read only archived will prevent creation of posts, so in this case it should error.

Reply:
Since we are doing first adding and then deleting, what you are saying makes a lot of sense. But what about the other direction?
What happens if we try to move (forget the UI, imagine I am pinging directly the REST API) from the read only / archived channel into a different channel?
I don't remember the actual logic you have here, but I could see something like "I start creating all the posts, and when I try to delete them, I have no permission, so it would end up as if I copied them over".

Is that scenario also handled?